resources/assets/components/CampaignIdSingle/index.js

```diff
@@ -15,13 +17,70 @@ class CampaignIdSingle extends React.Component {
     super(props);
 
     this.state = {
+      actions: {},
       displayCreateActionModal: false,
     };
 
+    this.api = new RogueClient(window.location.origin, {
+      headers: {
+        Authorization: `Bearer ${window.AUTH}`,
+      },
+    });
+
+    this.deleteAction = this.deleteAction.bind(this);
     // this.showCreateActionModal = this.showCreateActionModal.bind(this);
     // this.hideCreateActionModal = this.hideCreateActionModal.bind(this);
   }
 
+  componentDidMount() {
+    this.getActions(this.props.campaign.id);
+  }
+
+  /**
+   * Gets the campaign's actions.
+   *
+   * @param {Integer} campaignId
+   * @return {Object}
+   */
+  getActions(campaignId) {
+    this.api
+      .get(`api/v3/actions`, {
+        filter: {
+          campaign_id: campaignId,
+        },
+      })
+      .then(json =>
+        this.setState({
+          actions: keyBy(json.data, 'id'),
+        }),
+      );
+  }
+
+  // Delete an action.
+  deleteAction(actionId, event) {
+    event.preventDefault();
+    const confirmed = confirm(
+      '🚨🔥🚨 Are you sure you want to delete this? 🚨🔥🚨',
+    );
+
+    if (confirmed) {
+      // Make API request to Rogue to delete the action.
+      const response = this.api.delete(`actions/${actionId}`);
+
+      response.then(result => {
```

Comment:
Does this only happen if it was a good response?

Reply:
Yes! That is why I had to update the endpoint to return a response. It wasn't doing anything until I returned something and then when it got a good response, the state updated.